resilient_logger/utils.py

```diff
@@ -0,0 +1,146 @@
+import hashlib
+import json
+import logging
+from functools import cache
+from importlib import import_module
+from typing import Any, TypedDict, TypeVar
+
+from django.conf import settings
+from django.core.serializers.json import DjangoJSONEncoder
+
+from resilient_logger.abstract_log_source import AbstractLogSource
+from resilient_logger.missing_context_error import MissingContextError
+
+
+class ResilientLoggerConfig(TypedDict):
+    origin: str
+    environment: str
+    batch_limit: int
+    chunk_size: int
+    submit_unsent_entries: bool
+    clear_sent_entries: bool
+    sources: list[dict[str, Any]]
+    targets: list[dict[str, Any]]
+
+
+_default_config: ResilientLoggerConfig = {
+    "origin": "",
+    "environment": "",
+    "batch_limit": 5000,
+    "chunk_size": 500,
+    "clear_sent_entries": True,
+    "submit_unsent_entries": True,
+    "sources": [],
+    "targets": [],
+}
+
+BUILTIN_LOG_RECORD_ATTRS = {
+    "args",
+    "asctime",
+    "created",
+    "exc_info",
+    "exc_text",
+    "filename",
+    "funcName",
+    "levelname",
+    "levelno",
+    "lineno",
+    "module",
+    "msecs",
+    "message",
+    "msg",
+    "name",
+    "pathname",
+    "process",
+    "processName",
+    "relativeCreated",
+    "stack_info",
+    "taskName",
+    "thread",
+    "threadName",
+}
+
+TClass = TypeVar("TClass")
+
+
+def dynamic_class(type: type[TClass], class_path: str) -> type[TClass]:
+    """
+    Loads dynamically class of given type from class_path
+    and ensures it's sub-class of given input type.
+    """
+    parts = class_path.split(".")
+    class_name = parts.pop()
+    module_name = ".".join(parts)
+    module = import_module(module_name)
+    cls = getattr(module, class_name)
+
+    if not issubclass(cls, type):
+        raise Exception(f"Class '{class_path}' is not sub-class of the {type}.")
```

Comment:
```suggestion
        raise TypeError(f"Class '{class_path}' is not sub-class of the {type}.")
```
I guess it would be nicer to raise a more specific exception.